mods/default/trees.lua

```diff
@@ -426,3 +427,48 @@ function default.grow_new_aspen_tree(pos)
 	minetest.place_schematic({x = pos.x - 2, y = pos.y - 1, z = pos.z - 2},
 		path, "0", nil, false)
 end
+
+
+--
+-- Sapling 'on place' function to check protection of node and resulting tree volume
+--
+
+function default.sapling_on_place(itemstack, placer, pointed_thing,
+		sapling_name, minp_relative, maxp_relative, interval)
+	-- Position of sapling
+	local pos = pointed_thing.under
+	local node = minetest.get_node(pos)
```

Comment:
Nothing is impossible on laggy servers. Adding that ignore check will make it totally fail-safe.